proofs/flatten.go

```diff
@@ -185,29 +186,31 @@ func (f *messageFlattener) appendLeaf(prop Property, value string, salt []byte,
 	f.leaves = append(f.leaves, leaf)
 }
 
-func (f *messageFlattener) valueToString(value interface{}) (s string, err error) {
+func (f *messageFlattener) valueToBytesArray(value interface{}) (b []byte, err error) {
 	switch v := value.(type) {
 	case nil:
-		return "", nil
+		return []byte{}, nil
 	case string:
-		return v, nil
+		return []byte(v), nil
 	case int8, int16, int32, int64, uint8, uint16, uint32, uint64:
-		return fmt.Sprint(value), nil
+		return []byte(fmt.Sprint(v)), nil
 	case []byte:
-		return f.valueEncoder.EncodeToString(v), nil
+		return v, nil
 	case *timestamp.Timestamp:
 		if v == nil {
-			return "", nil
+			return []byte{}, nil
 		}
-		return ptypes.TimestampString(v), nil
+		return []byte(ptypes.TimestampString(v)), nil
 	default:
 		// special case for enums
 		rv := reflect.ValueOf(value)
 		if rv.Kind() == reflect.Int32 {
-			return strconv.FormatInt(rv.Int(), 10), nil
+			i := make([]byte, 8)
+			binary.LittleEndian.PutUint64(i, uint64(rv.Int()))
```

Comment:
We're using BigEndian in other places so let's use it here too: 

https://github.com/centrifuge/precise-proofs/blob/439d569e32536e2626d77eca0994d1544e8f4b36/proofs/property.go#L40